dissect/target/tools/diff.py

```diff
@@ -0,0 +1,961 @@
+
+        for differential in self.comparison.walkdir(path):
+            for entry, line in self._annotate_differential(
+                differential, include_unchanged, include_created, include_modified, include_deleted, absolute=True
+            ):
+                if not pattern.match(entry.name):
+                    continue
+
+                print(line, file=stdout)
+
+        return False
+
+    def do_plugin(self, line: str) -> bool:
+        """Yield RecordCreated, RecordUnchanged and RecordDeleted Records by comparing plugin outputs for two
+        targets."""
+        argparts = arg_str_to_arg_list(line)
+        pipeparts = []
+        if "|" in argparts:
+            pipeidx = argparts.index("|")
+            argparts, pipeparts = argparts[:pipeidx], argparts[pipeidx + 1 :]
+
+        if len(argparts) < 1:
+            raise ValueError("Provide a plugin name, and optionally parameters to pass to the plugin.")
+
+        plugin = argparts.pop(0)
+
+        iterator = self.comparison.differentiate_plugin_outputs(plugin, argparts)
+        if pipeparts:
+            try:
+                with build_pipe_stdout(pipeparts) as pipe_stdin:
+                    rs = RecordOutput(pipe_stdin.buffer)
+                    for record in iterator:
+                        rs.write(record)
+            except OSError as e:
+                # in case of a failure in a subprocess
+                print(e)
+        else:
+            for record in iterator:
+                print(record, file=sys.stdout)
+
+        return False
+
+    def do_python(self, line: str) -> bool:
+        """drop into a Python shell."""
+        python_shell(list(self.targets))
+        return False
+
+
+def make_target_pairs(targets: tuple[Target], absolute: bool = False) -> list[tuple[Target, Target]]:
+    """Make 'pairs' of targets that we are going to compare against one another. A list of targets can be treated in two
+    ways: compare every target with the one that came before it, or compare all targets against a 'base' target (which
+    has to be supplied as initial target in the list)."""
+    target_pairs = []
+
+    previous_target = targets[0]
+    for target in targets[1:]:
+        target_pairs.append((previous_target, target))
+        if not absolute:
+            # The next target should be compared against the one we just opened
+            previous_target = target
+    return target_pairs
+
+
+def differentiate_target_filesystems(
+    *targets: tuple[Target],
+    deep: bool = False,
+    limit: int = FILE_LIMIT,
+    absolute: bool = False,
+    include: list[str] = None,
+    exclude: list[str] = None,
+) -> Iterator[Record]:
+    """Given a list of targets, compare targets against one another and yield File[Created|Modified|Deleted]Records
+    indicating the differences between them."""
+    if len(targets) < 2:
+        raise ValueError("Provide two or more targets to differentiate between.")
```

Comment:
this is true for all modes right? maybe move it to a more generic place? - see my relevant corresponding suggestion...